r-reproducibility.Rmd

```diff
@@ -0,0 +1,1185 @@
+
+1. Create a new R Markdown file ("File -> New File -> R Markdown"), providing
+the title, author name (your name), and setting the output to HTML.
+2. Save the file as `using-rmarkdown.Rmd` in the `doc/` folder.
+3. Delete all text *except* the YAML header.
+4. Create an R code chunk and call the label "setup". Write code so the packages 
+and data are loaded.
+5. Create another code chunk and call it "plot-licenses-by-year". Write R code
+to create a point plot (`geom_point()`) of the year on the `x` axis and number
+of licenses on the `y` axis.
+6. Knit the document and see what the output looks like.
+7. Change the theme of the plot to another builtin theme (*hint*: themes start
+with `theme_`).
+
+#### Using R code chunks to insert tables
+
+You can also create tables by using the `kable()` function from the knitr package.
+Let's create a table of ...
+
+TODO: Finish this once data has been added.
+
+````markdown
+`r ''````{r mean-...-table}
+dog_licenses %>% 
+    select(...) %>% 
+    group_by(...) %>% 
+    summarise(Mean... = mean(..., na.rm = TRUE)) %>% 
+    spread(...) %>% 
+    knitr::kable(caption = "Table caption here.")
+`r ''````
+````
+
+```{r mean-...-table, echo=FALSE}
+dog_licenses %>% 
+    select(...) %>% 
+    group_by(...) %>% 
+    summarise(Mean... = mean(..., na.rm = TRUE)) %>% 
+    spread(...) %>% 
+    knitr::kable(caption = "Table caption here.")
+```
+
+If you want to simply print out a number or other text:
+
+````markdown
+`r ''````{r numbers-text}
+mean(dog_license$..., na.rm = TRUE)
+names(dog_license)
+`r ''````
+````
+
+```{r numbers-text, echo=FALSE}
+mean(dog_license$..., na.rm = TRUE)
+names(dog_license)
+```
+
+#### Inline R code
+
+Often you might have results inside the text you are writing. Here you can
+include R code within the text so that the results are inserted directly 
+into the document. It looks like:
+
+> The mean of BMI is &#96;r round(mean(NHANES$BMI, na.rm = TRUE), 2)&#96;.
+
+Which gives...
+
+> The mean of BMI is ` round(mean(NHANES$BMI, na.rm = TRUE), 2)`.
+
+Keep in mind that inline R code can *only* insert a single number or character
+value, nothing more.
+
+#### Citing literature with R Markdown
+
+No scientific writing is complete without being able to include references. If
+you want to insert a citation, use the Markdown key `[@Cone2016]`, which will
+look like [@Cone2016]. The text `Cone2016` is the key that the bibliography
+manager uses to identify a specific reference. Adding more references is done
+by separating by a `;`, so like `[@AuthorYear; @Author2Year; @Author3Year]`. The
+resulting citation reference will be inserted at the bottom of the document. To
+get the bibliography to work, you'll also need to add a line to the YAML header
+like this:
+
+```yaml
+---
+title: "My report"
+author: "Me!"
+bibliography: my_references.bib
+---
+```
+
+The `my_references.bib` is a `.bib` file found in the same folder as the `.Rmd` 
```

Comment:
Briefly mention that this file can be exported by common reference managers like Zotero and Mendeley so it is clear one does not need to create it by hand.